metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -497,7 +497,13 @@ def _get_column_cardinality(
         self, column_spec: _SingleColumnSpec, column: str
     ) -> None:
         try:
-            nonnull_count = self.dataset.get_column_nonnull_count(column)
```

Comment:
it would be nice to left some code comment on get_column_nonnull_count having issues and so using the new approach


I found the function being mentioned here, could you double check if this will have some unexpected side-effect?

https://github.com/datahub-project/datahub/blob/5ce93f54b40cdec5534ba3d878e6841065e83e66/metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py#L304-L312